src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,144 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as os from 'os';
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as mssql from 'vscode-mssql';
+import { defaultBindingResult, defaultSqlBindingTextLines, genericCollectionImport, sqlBindingResult } from '../constants/constants';
+import * as LocalizedConstants from '../constants/localizedConstants';
+import { AzureFunctionsService } from '../services/azureFunctionsService';
+import * as azureFunctionUtils from '../azureFunction/azureFunctionUtils';
+import * as constants from '../constants/constants';
+import * as path from 'path';
+
+export class AzureFunctionProjectService {
+
+	constructor(private azureFunctionsService: AzureFunctionsService) {
+	}
+
+	public async createAzureFunction(connectionString: string, schema: string, table: string): Promise<void> {
+		const azureFunctionApi = await azureFunctionUtils.getAzureFunctionsExtensionApi();
+		if (!azureFunctionApi) {
+			return;
+		}
+		if (!await isAzureFunctionProjectOpen()) {
+			vscode.window.showErrorMessage(LocalizedConstants.azureFunctionsProjectMustBeOpened);
+			return;
+		}
+
+		// because of an AF extension API issue, we have to get the newly created file by adding
+		// a watcher: https://github.com/microsoft/vscode-azurefunctions/issues/2908
+		const newFilePromise = getNewFunctionFile();
+
+		// get function name from user
+		const functionName = await vscode.window.showInputBox({
+			title: constants.functionNameTitle,
+			value: 'HttpTrigger1'
+		});
+
+		// create C# HttpTrigger
+		await azureFunctionApi.createFunction({
+			language: 'C#',
+			templateId: 'HttpTrigger',
+			functionName: functionName
+		});
+
+		// once steps are addressed we add SQL Bindings Nuget and Connection String to local.settings.json
+		// and then create a new functions file
+		await addNugetReferenceToProjectFile();
+		await addConnectionStringToConfig(connectionString);
+		const functionFile = await newFilePromise;
+
+
+		await this.azureFunctionsService.addSqlBinding(
+			mssql.BindingType.input,
+			functionFile,
+			functionName,
+			schema + escape('.') + table,
```

Comment:
This isn't what I meant by escape the string. You need to escape the object names to prevent possible SQL injection attacks (and allow names that have things like periods in them).

So escaping in this case is by bracket quoting the schema and table name (separately). So `dbo.myTable` becomes `[dbo].[myTable]`